lib/smart_proxy_remote_execution_ssh/multiplexed_ssh_connection.rb

```diff
@@ -0,0 +1,195 @@
+require 'smart_proxy_remote_execution_ssh/command_logging'
+
+module Proxy::RemoteExecution::Ssh::Runners
+  class SensitiveString
+    def initialize(value, mask: '*****')
+      @value = value
+      @mask = mask
+    end
+
+    def inspect
+      '"' + to_s + '"'
+    end
+
+    def to_s
+      @mask
+    end
+
+    def to_str
+      @value
+    end
+  end
+
+  class AuthenticationMethod
+    attr_reader :name
+    def initialize(name, prompt: nil, password: nil)
+      @name = name
+      @prompt = prompt
+      @password = password
+    end
+
+    def ssh_command_prefix
+      return [] unless @password
+
+      prompt = ['-P', @prompt] if @prompt
+      [{'SSHPASS' => SensitiveString.new(@password)}, '/usr/bin/sshpass', '-e', prompt].compact
```

Comment:
How is it ensured that `to_str` will be called on `SensitiveString` later instead of `to_s`?..

Reply:
Good question. When a hash is being printed it seems to be calling `#inspect` on the values and when the environment is being processed by `spawn` it seems to use `to_str` on them. Since `spawn` is implemented directly in C I did not dare to follow what it does and how it actually works under the hood/ It may prove insufficient if we later add more features, but for now it seems to work reasonably well

Reply:
ACK from me then :)